### PR TITLE
Enable ServiceAccount annotations

### PR DESCRIPTION
**Pull request by huang-jy**
This change request relates to https://github.com/aerokube/moon/issues/267 and will allow users to enable the moon service account to be used with Workload Identity (https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) or allow users to tag the service account with custom annotations which could be used for security or monitoring.

The annotations are not mandatory and no Moon functionality is changed by this PR being (or not being) merged.

One note: The current chart version available on the helm repo is 1.1.4, but the Chart.yaml in the git repo shows 1.0.7, I bumped the version to 1.1.5 for this change. If that is wrong, let me know and I'll put it down to 1.0.8

### CODE REVIEW

File: moon/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-name: moon
-version: 1.0.7
 appVersion: 1.8.3
 description: Moon Helm chart
+name: moon
+version: 1.1.5
```
Comment (vania-pooh) [inline moon/Chart.yaml:5]: This change makes is useless. Version is being set automatically when deploying chart: https://github.com/aerokube/charts/blob/master/ci/build.sh#L13.

File: moon/templates/moon.yaml [hunk 2]
```diff
@@ -4,6 +4,12 @@ kind: ServiceAccount
 metadata:
   name: {{ .Release.Name }}
   namespace: {{ .Release.Namespace }}
+  {{- if .Values.moon.serviceaccount.annotations }}
```
Comment (vania-pooh) [inline moon/templates/moon.yaml:7]: Need to check for null value:
```
{{- if and .Values.moon.serviceaccount .Values.moon.serviceaccount.annotations }}
```

**Review by vania-pooh — CHANGES_REQUESTED** [review]
Please apply two straightforward fixes.

File: moon/templates/moon.yaml (hunk 2 above)
Comment (huang-jy) [reply moon/templates/moon.yaml:7]: Ah yes, good catch

**Review by vania-pooh — APPROVED** [review]

**Comment by vania-pooh** [comment]
@huang-jy thank you for contribution. These changes were automatically pushed to `aerokube-unstable` repository described in README.md. Would you need a stable version now?

**Comment by huang-jy** [comment]
I can use the unstable version for testing the change. Will the change make
it into stable eventually? Or do I need to trigger something for that to
happen?

**Comment by vania-pooh** [comment]
@huang-jy we need to create a tag in this repo.

**Comment by huang-jy** [comment]
Thanks, I'll test the change on the unstable, then let you know if it
behaves as expected. If so, we can then create the tag, or do whatever else
is necessary.

**Comment by huang-jy** [comment]
Test successful. When installing the chart from stable with the annotation set to work with workload identity, then trying to use the workload test, it failed.

When installing from unstable, then running the same test, it worked.

**Pull request merged**